backend/src/main/java/com/htecgroup/skynest/controller/UserController.java

```diff
@@ -258,4 +262,46 @@ public ResponseEntity<String> deleteUser(@PathVariable UUID uuid) {
     log.info(deleteSuccess);
     return ResponseEntity.ok(deleteSuccess);
   }
+
+  @Operation(summary = "Request for token refresh")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "new access token successfully sent",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = String.class),
+                  examples = {@ExampleObject(value = "Password reset email sent")})
+            }),
+        @ApiResponse(
+            responseCode = "403",
+            description = "Access denied",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = String.class),
+                  examples = {@ExampleObject(value = "Access token is invalid")})
+            }),
+        @ApiResponse(
+            responseCode = "500",
+            description = "Internal Server Error",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = String.class),
+                  examples = {
+                    @ExampleObject(value = "Failed to send new access and refresh tokens")
+                  })
+            })
+      })
+  @GetMapping("/token/refresh")
+  public void refreshToken(
+      @RequestHeader("refresh-token") String refresh_token, HttpServletResponse response) {
```

Comment:
I'm not sure this header will go through CORS.

Also, how will this request pass through our authorization filter? The access token is already expired.

Reply:
This must be done while access token is still active, if not it can't pass you're right.

Reply:
I guess you could just expect the refresh token in the authorization header here. Then it would just magically work with both CORS and the authorization filter. Not sure the mentors would like that though.

Reply:
you are right , I don't like that hack at all :) it will work sure, but it is not the correct approach..

Reply:
I will try to answer that in the PR it self